.changes/unreleased/FEATURES-20240404-154553.yaml

```diff
@@ -0,0 +1,8 @@
+kind: FEATURES
+body: 'types/validation: Added `BoolValuableWithValidateableAttribute`, `Float64ValuableWithValidateableAttribute`,
```

Comment:
The utility of the value-type interfaces is debatable as the interface that they embed, `ValidateableAttribute`, can be implemented directly by a custom value type. However, the value-type interfaces do streamline defining the interfaces that a custom value type should implement. For example

Using `xattr.ValidateableAttribute`:

```go
var _ basetypes.StringValuable = CustomStringValue{}
var _ xattr.ValidateableAttribute = CustomStringValue{}
```

Using `xattr. StringValuableWithValidateableAttribute`:
```go
var _ StringValuableWithValidateableAttribute = CustomStringValue
```

Reply:
Currently, I'm leaning towards omitting the `{TYPE}ValuableWithValidatableAttribute` and `{TYPE}ValuableWithValidatableParameter` interfaces.

In the case of implementing all 3 interfaces you'd potentially have something like:
```go
var (
   _ validation.StringValuableWithValidateableAttribute = CustomStringValue
   _ validation.StringValuableWithValidateableParameter = CustomStringValue
)
```
vs.
```go
var (
   _ basetypes.StringValuable = CustomStringValue
   _ xattr.ValidateableAttribute = CustomStringValue
   _ function.ValidateableParameter = CustomStringValue
)
```

I think the second is a little more readable and explicit

Reply:
I think that `{TYPE}ValuableWithValidatableAttribute` and `{TYPE}ValuableWithValidatableParameter` interfaces are easier to discover with the clear naming, but since existing custom value types already implement `basetypes.{Type}Valuable`, I think it might be confusing to remove that interface just to obfuscate it behind another interface.

Reply:
Thanks for the feedback. I've removed the `<Type>ValuableWithValidateable<Attribute|Parameter>` interfaces.